keygen/src/keygen.rs

```diff
@@ -729,6 +741,20 @@ fn do_main(matches: &ArgMatches) -> Result<(), Box<dyn error::Error>> {
             };
             let no_outfile = matches.is_present(NO_OUTFILE_ARG.name);
 
+            // The vast majority of base58 encoded public keys have length 44, but
+            // these only enacapsulate prefixes 1-9 and A-H.  If the user is searching
+            // for a keypair that starts with a prefix of J-Z or a-z, then there is no
+            // reason to waste time searching for a keypair that will never match
+            let skip_len_44_pubkeys = grind_matches
```

Comment:
it'd probably pay to warn the user that they're searching for an unlikely prefix

Reply:
Majority of prefixes are pretty unlikely :)

Reply:
Looks like this does not take into account the `--ignore-case` argument

Reply:
> Looks like this does not take into account the `--ignore-case` argument

https://github.com/solana-labs/solana/pull/30365/files#r1158852602